prow/plugins/config.go

```diff
@@ -398,7 +398,10 @@ type ConfigMapSpec struct {
 	// Namespaces in which the configMap needs to be deployed, in addition to the above
 	// namespace provided, or the default if it is not set.
 	AdditionalNamespaces []string `json:"additional_namespaces,omitempty"`
-
+	// GZIP toggles whether the key's data should be GZIP'd before being stored
+	// If set to false and the global GZIP option is enabled, this file will
+	// will not be GZIP'd.
+	GZIP *bool `json:"gzip,omitempty"`
```

Comment:
I don't think we need a pointer, right? Unset here is `false` which is the current default

Reply:
We do, to allow the global override to work. We need to distinguish between "explicitly false" and "implicitly false".

Similar to `deployment.spec.replicas` which is a *int32 in Kubernetes - `replicas: 0` means something different to omitting the replicas field altogether 😄

Reply:
> Unset here is false which is the current default

To clarify my last comment, unset here is *not* false, it is `nil` :). False means false, nil means "not specified"

Reply:
Hrm. Do we have any real use cases for a global default? Does any extant deployment of Prow have any chance of using the global default? I understand it's the current approach but it does seem to make this more complex than it should be and unless we have a use-case for it, perhaps we should just omit it ...

Reply:
I am not sure if I'd say this is the most complex part of Prow... it boils down to 4 lines of code that actually need to handle this: https://github.com/kubernetes/test-infra/pull/11765/files#diff-ac6db59f9774f9c51329fc7a401ae966R206

This could easily be moved out to a standalone function in future if any other code needs to build on this plugin's config.

If we *don't* keep this as a *bool, then a config like this:

```yaml
config_updater:
  gzip: true
  maps:
    label_sync/labels.yaml:
      name: label-config
      gzip: false
    prow/config.yaml:
      name: config
    prow/plugins.yaml:
      name: plugins
    config/jobs/**/*.yaml:
      name: job-config
```

would result in *all* files being gzipped, despite the labels.yaml entry explicitly setting `gzip: false`.

IMO, this will confuse users.

> Does any extant deployment of Prow have any chance of using the global default?

prow.k8s.io needs to do this today, or otherwise needs to stop using the global default and instead set `gzip: true` explicitly on 3 other entries.

I'm not really too sure what your concern here is - it's not like the updateconfig config surface is widely used in *other* areas of prow, so there's really not many places that ever need to evaluate this, and even if there are it's a pretty simple algorithm... "use specific gzip config if set, otherwise use global default" 🙄

Reply:
I'm saying to remove the global, because the config with a simple opt-in per config would be 

```yaml
config_updater:
  maps:
    label_sync/labels.yaml:
      name: label-config
    prow/config.yaml:
      name: config
    prow/plugins.yaml:
      name: plugins
    config/jobs/**/*.yaml:
      name: job-config
      gzip: true
```

You only really need job config to be gzipped.

We don't have a need for a global flag, my biggest worry with `nil` values is we end up getting caught by them at all points -- every time we had a `nil` pop up in plugin config over the years we've had numerous fixes and re-fixes to handle panics. Also, the changes to not being stringable, etc, also stem from the pointer.

/shrug

I am well aware of and fully understand how to use a pointer and `omitempty` to tell if a value is set via config, I'm just not convinced it's useful. Let me review the code again to see how much of the added lines here would change because of it